scripts/partner_batch_imports.py

```diff
@@ -0,0 +1,197 @@
+"""
+Process partner bibliographic csv data into importable json book
+records and then batch submit into the ImportBot
+`import_item` table (http://openlibrary.org/admin/imports)
+which queues items to be imported via the
+Open Library JSON import API: https://openlibrary.org/api/import
+"""
+
+import os
+import re
+import sys
+import web
+import datetime
+from datetime import timedelta
+import logging
+import requests
+
+# Add openlibrary into our path so we can process config + batch functions
+from openlibrary.core.imports import Batch
+from infogami import config
+from openlibrary.config import load_config
+
+logger = logging.getLogger("openlibrary.importer.bwb")
+
+SCHEMA_URL = "https://raw.githubusercontent.com/internetarchive" \
+             "/openlibrary-client/master/olclient/schemata/import.schema.json"
+
+
+class Biblio():
+
+    ACTIVE_FIELDS = [
+        'title', 'isbn_13', 'publish_date', 'publishers',
+        'weight', 'authors', 'lc_classifications', 'number_of_pages',
+        'languages', 'subjects', 'source_records'
+    ]
+    INACTIVE_FIELDS = [
+        "copyright", "issn", "doi", "lccn", "dewey", "length",
+        "width", "height"
+    ]
+    REQUIRED_FIELDS = requests.get(SCHEMA_URL).json()['required']
+
+    def __init__(self, data):
+        self.isbn = data[124]
+        self.isbn_13 = [self.isbn]
+        self.title = data[10]
+        self.publish_date = data[20][:4]  # YYYY, YYYYMMDD
+        self.publishers = [data[135]]
+        self.weight = data[39]
+        self.authors = self.contributors(data)
+        self.lc_classifications = data[147]
+        self.number_of_pages = next(
```

Comment:
@mekarpeles this has come up before of https://github.com/internetarchive/openlibrary/issues/4204 , but the `data[36]` field should correspond to the OL text field `pagination` , not the int `number_of_pages`.

From memory, the consensus amongst librarian users was that the `pagination` field is more useful, and generally we only get an integer count of number of pages from the IA scanning process. This regex method looks lossy and and is going to undercount pages in many cases, and lose part of the provided pagination information.